lib/internal/http2/core.js

```diff
@@ -1663,12 +1667,12 @@ class ClientHttp2Session extends Http2Session {
     this[kUpdateTimer]();
 
     if (headers !== null && headers !== undefined) {
-      for (const header of ObjectKeys(headers)) {
+      ArrayPrototypeForEach(ObjectKeys(headers), (header) => {
```

Comment:
`for..of` loop has higher performance as compared to `forEach` (correct me if I am wrong)...
So are we still fine going with `forEach` for this specific case here?

P.S. Just started with open-source reviews and was trying to learn from other PRs.

Reply:
Yes, `for..of` is more performant which is why it was chosen to begin with. It's also why this pull request will need a benchmark run before it can be merged. The problem with `for..of` is that it is vulnerable to prototype pollution attacks. If some third party module deep in your dependencies decides to inject some code in `Array.prototype[Symbol.iterator]`, then `for..of` will happily run that code. That may or may not be a problem for the end user's code--there are legitimate use cases (like profiling) for this sort of so-called "monkeypatching", but we don't want Node.js core to run the altered code.